src/asm/validate_header.c

```diff
@@ -0,0 +1,116 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   validate_header.c                                  :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: asarandi <asarandi@student.42.us.org>      +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2018/02/11 20:41:20 by asarandi          #+#    #+#             */
+/*   Updated: 2018/02/28 01:35:18 by asarandi         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "asm.h"
+
+void	validate_champion_name(t_asm *a, int *i)
+{
+	int		len;
+	char	*quote1;
+	char	*str;
+
+	str = ft_strdup(a->split[*i]);
+	while ((count_matching_chars(str, '"') < 2) && (*i < a->lines - 1))
+		append_nl_string(&str, a->split[++*i]);
+	if (count_matching_chars(str, '"') > 2)
+		return (header_error(".name has too many quote marks (\")", a));
+	if (count_matching_chars(str, '"') == 2)
+	{
+		quote1 = ft_strchr(str, '"') + 1;
+		len = ft_strchr(quote1, '"') - quote1;
+		if (len < 1)
+			warning(".name is blank");
+		if (len > PROG_NAME_LENGTH)
+			return (header_error(".name length exceeds PROG_NAME_LENGTH", a));
+		ft_strncpy(a->header.prog_name, quote1, len);
```

Comment:
:warning: ft_strncpy() takes a size_t as its third argument, which is unsigned! A negative length will only print a warning and then proceed to copy a gigantic number of characters.

This appears to be prevented from actually happening via the count_matching_chars() call but that's a pretty nasty precondition to prevent a potential security bug.

Reply:
casting len to (int) .. however negative length should be impossible